casepro/statistics/models.py

```diff
@@ -122,6 +124,30 @@ class Meta:
         abstract = True
 
 
+class BaseAverage(BaseCount):
+    """
+    Tracks total minute counts of different items (e.g. time since assigned ) in different scopes (e.g. org, user)
+    """
+    TYPE_TILL_REPLIED = 'A'
+    TYPE_TILL_CLOSED = 'C'
+
+    squash_sql = """
+        WITH removed as (
+            DELETE FROM %(table_name)s WHERE %(delete_cond)s RETURNING "count", "total"
+        )
+        INSERT INTO %(table_name)s(%(insert_cols)s, "count", "total")
+        VALUES (
+            %(insert_vals)s,
+            GREATEST(0, (SELECT SUM("count") FROM removed)),
+            GREATEST(0, (SELECT SUM("total") FROM removed))
```

Comment:
It's possible that we might be wanting to track negative averages. Perhaps we should use `COALESCE` here instead.